src/addons/jslider.cpp

```diff
@@ -60,8 +62,8 @@ void JSliderInput::process()
 #endif
 
     Gamepad * gamepad = Storage::getInstance().GetGamepad();
-    if ( gamepad->options.dpadMode != dpadState) {
-        gamepad->options.dpadMode = dpadState;
+    if ( gamepad->getOptions().dpadMode != dpadState) {
+        gamepad->getOptions().dpadMode == dpadState;
```

Comment:
LD/DP/RS selection doesn't seem to do anything anymore, I think line 66 here is errant and the cause.

Reply:
This fixes it:

```
diff --git a/headers/gamepad.h b/headers/gamepad.h
index 0328c9c..bbaf85f 100644
--- a/headers/gamepad.h
+++ b/headers/gamepad.h
@@ -126,6 +126,7 @@ public:
 
 	const GamepadOptions& getOptions() const { return options; }
 
+	void setDpadMode(DpadMode dpadMode) { options.dpadMode = dpadMode; }
 	void setInputMode(InputMode inputMode) { options.inputMode = inputMode; }
 	void setSOCDMode(SOCDMode socdMode) { options.socdMode = socdMode; }
 
diff --git a/src/addons/jslider.cpp b/src/addons/jslider.cpp
index 04ab776..d30302e 100644
--- a/src/addons/jslider.cpp
+++ b/src/addons/jslider.cpp
@@ -63,7 +63,7 @@ void JSliderInput::process()
 
     Gamepad * gamepad = Storage::getInstance().GetGamepad();
     if ( gamepad->getOptions().dpadMode != dpadState) {
-        gamepad->getOptions().dpadMode == dpadState;
+        gamepad->setDpadMode(dpadState);
         gamepad->save();
     }
 }
```

Reply:
@bsstephan I just submitted a near identical fix :)

Good catch!